pkg/helpers/newapp/app/imageref.go

```diff
@@ -230,18 +239,25 @@ func (r *ImageRef) SuggestName() (string, bool) {
 	if r == nil {
 		return "", false
 	}
+	var name string
 	// if ImageStream already exists, its name should
 	// have higher preference than object name
-	if r.Stream != nil {
-		return r.Stream.Name, true
-	}
-	if len(r.ObjectName) > 0 {
-		return r.ObjectName, true
-	}
-	if len(r.Reference.Name) > 0 {
-		return r.Reference.Name, true
+	switch {
+	case r.Stream != nil:
+		name = r.Stream.Name
+	case len(r.ObjectName) > 0:
+		name = r.ObjectName
+	case len(r.Reference.Name) > 0:
+		name = r.Reference.Name
 	}
-	return "", false
+
+	// in some certain situations, ref name might contain slashes or other symbols
+	// k8s naming goes along rfc952, which states that only alphanumeric characters,
+	// dots and dashes are allowed in a name
+	// bug example: https://bugzilla.redhat.com/show_bug.cgi?id=1970805
+	name = dnsNamePattern.ReplaceAllString(name, "-")
```

Comment:
I would swear that we have existing code somewhere that does this (maybe not in this repository). Maybe in library-go?